app/src/main/java/com/alphawallet/app/ui/NewSettingsFragment.java

```diff
@@ -334,8 +349,19 @@ private void addSettingsToLayout()
 
         walletSettingsLayout.addView(walletConnectSetting, walletIndex++);
 
-        if (CustomViewSettings.getLockedChains().size() == 0)
-            systemSettingsLayout.addView(selectNetworksSetting, systemIndex++);
+
+        try {
+            String lockedChains = Utils.loadJSONStringFromAsset(getContext(), Utils.CUSTOM_SETTINGS_FILENAME);
+            JSONObject customSettingsJsonObject = new JSONObject(lockedChains);
+            JSONArray chainsArray = customSettingsJsonObject.getJSONArray("locked_chains");
+            if (chainsArray.length() == 0)
+            {
+                systemSettingsLayout.addView(selectNetworksSetting, systemIndex++);
+            }
+        }catch (JSONException err){
+            err.printStackTrace();
+        }
+
 
```

Comment:
FORMATTING. USE THE ALPHAWALLETCONFIG.XML

Also, chains handling code should be common in the config file. You may need to refactor the custom view settings file so it acts as an instance of the viewmodel class that is using it, that way it can use the context of the hosting class, and be sure the context pointer of the custoviewsettings class will be destroyed before the host context is.